### DISCUSSION TITLE
localPosition gets NaN error in ScrollSnap

### DISCUSSION THREAD

**Discussion by Qualdan** [Unity-UI-Extensions/com.unity.uiextensions#452] [Q&A]
I'm trying to use ScrollSnap, but it produces error:
`transform.localPosition assign attempt for 'Content' is not valid. Input localPosition is { 0.000000, NaN, 0.000000 }.`

This happens in:
**ScrollSnap.cs** > **LateUpdate()** on the line
`_listContainerTransform.localPosition = Vector3.Lerp(_listContainerTransform.localPosition, _lerpTarget, 7.5f * Time.deltaTime);`

I have vertical _Scrollview_ with _Vertical Layout Group_ and _Content Size Fitter_ on _Content_. I also have buttons for _Up_ and _Down_ movement.

I get this error every time if I'm at the top, click down button and then try to scroll up (or the reverse, starting on the bottom, clicking up button and scrolling down). I can give more information if this isn't enough.

**Comment by SimonDarksideJ** [comment]
Sorry for the late reply, notifications are still not working, gah.

Can you provide a sample scene just using the control that replicates the behaviour please @Qualdan , so I can better test a fix should one be required,

**Reply by Qualdan** [reply]
Hi. Thanks for the reply.

The weird thing is that on an empty project I don't get the exact same problem, but a similar one.

On the included test project (SnapTest.zip) click the "Down" button and then try to scroll back up with a mouse wheel. It won't let you. You have to manually move the scrollbar to "unstick" the scrolling.

In the original problem I got an error while first using the buttons and then trying to scroll. I'm not sure if these are related problems.

The test project was too large to include here, so I put it in my Dropbox:
https://www.dropbox.com/scl/fi/mul8qo9byd06gecacn5ya/SnapTest.zip?rlkey=8hf28zq991fp46kkwikmc95md&dl=0

**Reply by SimonDarksideJ** [reply]
Thanks, will be sure to look into this for the Unity 6 update

**Reply by SimonDarksideJ** [reply]
Added t the Unity 6 workoff list
https://github.com/Unity-UI-Extensions/com.unity.uiextensions/issues/498#issue-3670852294

**Comment by SimonDarksideJ** [comment]
Resolved in #508
